astrowidgets/glupyter.py

```diff
@@ -0,0 +1,400 @@
+            Extension number of the desired HDU. If not given, it is
+            assumed to be in EXT 0.
+
+        memmap : bool or `None`
+            Memory mapping. See `astropy.io.fits.open`.
+
+        """
+        with fits.open(filename, memmap=memmap) as pf:
+            data = _new_data_from_hdulist(pf, numhdu)
+
+        viewer = self._viewer
+        data = validate_data_argument(self._app.data_collection, data)
+
+        # NOTE: Changing state via API does not work properly if we loaded
+        # multiple data into the same widget in the same session.
+        self._app.data_collection.clear()
+
+        self._app.add_datasets(data)
+        viewer.add_data(data)
+        viewer.state.reference_data = data  # Bring it to focus
+        self._link_image_to_cb()
+
+    def load_nddata(self, nddata, label=None):
+        """Load a `~astropy.nddata.NDData` object into the viewer.
+
+        .. todo:: Add flag/masking support, etc.
+
+        Parameters
+        ----------
+        nddata : `~astropy.nddata.NDData`
+            ``NDData`` with image data and WCS.
+
+        label : str or `None`
+            Glue label name for the data.
+
+        """
+        if label is None:
+            label = _gen_random_label()
+
+        data = Data(label=label)
+
+        if nddata.wcs is not None:
+            coords = coordinates_from_wcs(nddata.wcs)
+            data.coords = coords
+
+        data.add_component(component=nddata.data, label=label)
+
+        viewer = self._viewer
+        data = validate_data_argument(self._app.data_collection, data)
+
+        # NOTE: Changing state via API does not work properly if we loaded
+        # multiple data into the same widget in the same session.
+        self._app.data_collection.clear()
+
+        self._app.add_datasets(data)
+        viewer.add_data(data)
+        viewer.state.reference_data = data  # Bring it to focus
+        self._link_image_to_cb()
+
+    def load_array(self, arr, label=None):
+        """Load a 2D array into the viewer.
+
+        .. note:: Use :meth:`load_nddata` for WCS support.
+
+        Parameters
+        ----------
+        arr : array-like
+            2D array.
+
+        label : str or `None`
+            Glue label name for the data.
+
+        """
+        if label is None:
+            label = _gen_random_label()
+
+        data = Data(label=label)
+        data.add_component(component=arr, label=label)
+
+        viewer = self._viewer
+        data = validate_data_argument(self._app.data_collection, data)
+
+        # NOTE: Changing state via API does not work properly if we loaded
+        # multiple data into the same widget in the same session.
+        self._app.data_collection.clear()
+
+        self._app.add_datasets(data)
+        viewer.add_data(data)
+        viewer.state.reference_data = data  # Bring it to focus
+        self._link_image_to_cb()
+
+    def center_on(self, point):
+        raise NotImplementedError  # FIXME
```

Comment:
You can find and set the current limits with ``viewer.state.x_min``, ``viewer.state.x_max``, ``viewer.state.y_min`` and ``viewer.state.y_max`` so you could use that to compute dx and dy (the field of view) then set e.g. ``viewer.state.x_min = point[0] - dx/2`` and so on if ``point`` is in pixel coordinates

Reply:
Given that the viewer size and display scale is already known, is it sufficient if I just set `x_min` and `y_min` and then let glupyter take care of recomputing the rest, or is that not a thing?

I am guessing, it takes `x/y_min/max` that is out of range? Otherwise, I cannot center on edges.

Reply:
The math isn't quite right. It slowly zooms the image out.

Reply:
Okay. I think it is because I am setting only `x_min` and `y_min` and let it auto adjust for the max. Is that not the right way to do this?